evap/staff/views.py

```diff
@@ -1491,8 +1491,18 @@ def notify_reward_points(grantings, **_kwargs):
     user = get_object_or_404(UserProfile, id=user_id)
     form = UserForm(request.POST or None, request.FILES or None, instance=user)
 
-    semesters_with_evaluations = Semester.objects.filter(courses__evaluations__contributions__contributor=user).distinct()
-    evaluations_contributing_to = [(semester, Evaluation.objects.filter(course__semester=semester, contributions__contributor=user)) for semester in semesters_with_evaluations]
+    semesters_with_evaluations = Semester.objects.filter(
+        Q(courses__evaluations__contributions__contributor=user) | Q(courses__responsibles__in=[user])
+    ).distinct()
+    evaluations_contributing_to = [
+        (
+            semester,
+            Evaluation.objects.filter(
+                Q(course__semester=semester) & (Q(contributions__contributor=user) | Q(course__responsibles__in=[user]))
+            ).distinct()
+        )
+        for semester in semesters_with_evaluations
+    ]
```

Comment:
I would suggest to only do the evaluations query, and then group by semesters manually.

Also, this does not match the commit message.